File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.py
class Solution:
    def removeOuterParentheses(self, s: str) -> str:
        from collections import deque
        newS = ""
        queue = deque()
        leftParenthesis = 0
        for char in s:
            if char == "(":
                queue.append(char)
                leftParenthesis+=1
            elif char == ")" and leftParenthesis == 1:
                leftParenthesis = 0
                queue.popleft()
                newS += "".join(queue)
                queue.clear()
            else:
                queue.append(char)
                leftParenthesis -= 1
        return newS

File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.java
class Solution {
    public String removeOuterParentheses(String s) {
        String newS = "";
        int leftParenthesis = 0;
        for(int i = 0; i<s.length();i++){
            char c = s.charAt(i);
            if(c == '(' && leftParenthesis == 0) {
                leftParenthesis += 1;
            } else if(c == '(' && leftParenthesis >= 1) {
                newS += c;
                leftParenthesis += 1;
            } else if(c==')' && leftParenthesis == 1){
                leftParenthesis = 0;
            } else {
                newS += c;
                leftParenthesis -= 1;
            }
        }
        return newS;
    }
}